orthoinference/src/main/java/org/reactome/orthoinference/InferEvents.java

```diff
@@ -0,0 +1,307 @@
+			for (GKInstance reactionInst : reactionInstances) {
+				dbids.add(reactionInst.getDBID());
+				reactionMap.put(reactionInst.getDBID(), reactionInst);
+			}
+			// For now sort the instances by DB ID so that it matches the Perl sequence
+			Collections.sort(dbids);
+			
+			if (!reactionInstances.isEmpty())
+			{
+				for (Long dbid : dbids)
+				{
+					GKInstance reactionInst = reactionMap.get(dbid);
+					// Check if the current Reaction already exists for this species, that it is a valid instance (passes some filters), and that it doesnt have a Disease attribute.
+					// Adds to manualHumanEvents array if it passes conditions. This code block allows you to re-run the code without re-inferring instances.
+					ArrayList<GKInstance> previouslyInferredInstances = new ArrayList<GKInstance>();
+					for (GKInstance orthoEventInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList(ReactomeJavaConstants.orthologousEvent))
+					{
+						GKInstance reactionSpeciesInst = (GKInstance) orthoEventInst.getAttributeValue(ReactomeJavaConstants.species);
+						if (reactionSpeciesInst.getDBID() == speciesInst.getDBID() && orthoEventInst.getAttributeValue(ReactomeJavaConstants.isChimeric) == null)
+						{
+							previouslyInferredInstances.add(orthoEventInst);
+							
+						}
+					}
+					for (GKInstance inferredFromInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList(ReactomeJavaConstants.inferredFrom))
+					{
+						GKInstance reactionSpeciesInst = (GKInstance) inferredFromInst.getAttributeValue(ReactomeJavaConstants.species);
+						if (reactionSpeciesInst.getDBID() == speciesInst.getDBID() && inferredFromInst.getAttributeValue(ReactomeJavaConstants.isChimeric) == null)
+						{
+							previouslyInferredInstances.add(inferredFromInst);
+						}
+					}
+					if (previouslyInferredInstances.size() > 0)
+					{
+						GKInstance prevInfInst = previouslyInferredInstances.get(0);
+						if (prevInfInst.getAttributeValue(ReactomeJavaConstants.disease) == null)
+						{
+							manualEventToNonHumanSource.put(reactionInst, prevInfInst);
+							manualHumanEvents.add(reactionInst);
+						} else {
+							System.out.println("Skipping building of hierarchy around pre-existing disease reaction " + prevInfInst);
+						}
+						
+						continue;
+					}
+					// This Reaction doesn't already exist for this species, and an orthologous inference will be attempted.
+						InferReaction.reactionInferrer(reactionInst);
+				}
+			}
+		}
+		UpdateHumanEvents.setInferredEvent(InferReaction.getInferredEvent());
+		UpdateHumanEvents.updateHumanEvents(InferReaction.getInferrableHumanEvents());
+		InferEvents.outputReport(species);
+		InferEvents.resetVariables();
+//		System.gc();
+		System.out.println("Finished orthoinference of " + speciesName + ".");
+		}
+	}
+
+	public static void outputReport(String species) throws IOException
+	{
+		int[] counts = InferReaction.getCounts();
```

Comment:
It's probably clearer to separate this into two getters rather than returning an array.
`InferReaction.getEligibleCount()` and `InferReaction.getInferredCount()`